crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -2410,8 +2420,13 @@ pub(crate) mod tests {
 
     #[async_test]
     async fn test_create_olm_machine() {
+        let test_start_ts = MilliSecondsSinceUnixEpoch::now();
         let machine = OlmMachine::new(user_id(), alice_device_id()).await;
 
+        let device_creation_time = machine.device_creation_time();
+        assert!(device_creation_time <= MilliSecondsSinceUnixEpoch::now());
+        assert!(device_creation_time >= test_start_ts);
```

Comment:
(no action needed) SystemTime (used by `MilliSecondsSinceUnixEpoch`) isn't monotonic, let's hope it's monotonic enough for these checks to not fail intermittently :)